src/picongpu/include/simulation_defines/param/speciesDefinition.param

```diff
@@ -143,7 +172,21 @@ PIC_Ions
 typedef MakeSeq<
 Species1,
 Species2
+>::type VectorSpeciesTmp;
+
+// If there are species containing the `synchrotronPhotons` alias, add photon species to `VectorAllSpecies`
+typedef typename PMacc::particles::traits::FilterByFlag<
```

Comment:
This reason why this not compile is that `FilterByFlag` is unknown at this point.

output of the compile suite:
```
>  /home/buildbot/build/params/KelvinHelmholtz/cmakePreset_4/include/simulation_defines/param/speciesDefinition.param(178): error: name followed by "::" must be a class or namespace name
>  
>  /home/buildbot/build/params/KelvinHelmholtz/cmakePreset_4/include/simulation_defines/param/speciesDefinition.param(179): error: expected an identifier
>  
>  /home/buildbot/build/params/KelvinHelmholtz/cmakePreset_4/include/simulation_defines/param/speciesDefinition.param(180): error: an explicit template argument list is not allowed on this declaration
>  
>  /home/buildbot/build/params/KelvinHelmholtz/cmakePreset_4/include/simulation_defines/param/speciesDefinition.param(181): error: expected a ";"
>  
>  /home/buildbot/build/params/KelvinHelmholtz/cmakePreset_4/include/simulation_defines/param/speciesDefinition.param(184): error: identifier "AllSynchrotronPhotonsSpecies" is undefined
>  
>  /home/buildbot/build/params/KelvinHelmholtz/cmakePreset_4/include/simulation_defines/param/speciesDefinition.param(188): error: identifier "PIC_Photons" is undefined
>  
```